### PR TITLE
Event 586 facebook instagram logout

### PR DESCRIPTION
**Pull request by katefree**

### CODE REVIEW

File: app/scripts/services/logout.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    $sce,
+  ) {
+    this.logoutFormProviders = function(response) {
+      switch ($cookies.get('crsAuthProviderType')) {
+        case 'FACEBOOK':
+          // if facebook, then logout from facebook using url generated on the backend
+          if (response.data) {
+            $window.location.href = response.data.url;
+          }
+          break;
```
Comment (TheNoodleMoose) [inline app/scripts/services/logout.js]: You could destructure the response to just grab data and then make it an arrow function as well.
```suggestion
    this.logoutFormProviders = ({ data }) => {
      switch ($cookies.get('crsAuthProviderType')) {
        case 'FACEBOOK':
          // if facebook, then logout from facebook using url generated on the backend
          if (data) {
            $window.location.href = data.url;
          }
          break;
```

File: test/spec/services/logout.spec.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import 'angular-mocks';
+
+describe('Service: logoutService', function() {
```
Comment (TheNoodleMoose) [inline test/spec/services/logout.spec.js]: Can you make this an arrow function?

File: test/spec/services/logout.spec.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  let logoutService, $rootScope, $window, $sce;
+  let cookieValue = null;
+
+  beforeEach(
+    angular.mock.module('confRegistrationWebApp', function($provide) {
```
Comment (TheNoodleMoose) [inline test/spec/services/logout.spec.js]: and here

File: test/spec/services/logout.spec.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      $provide.value('$window', { location: { href: 'dummy' } });
+      $provide.value('$cookies', {
+        get: function() {
```
Comment (TheNoodleMoose) [inline test/spec/services/logout.spec.js]: Here

File: test/spec/services/logout.spec.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+          return cookieValue;
+        },
+      });
+    }),
+  );
+
+  beforeEach(inject(function(_logoutService_, _$rootScope_, _$sce_, _$window_) {
```
Comment (TheNoodleMoose) [inline test/spec/services/logout.spec.js]: Here

File: test/spec/services/logout.spec.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    $rootScope = _$rootScope_;
+    $sce = _$sce_;
+    $window = _$window_;
+    logoutService = _logoutService_;
+  }));
+
+  it('should logout from facebook', function() {
```
Comment (TheNoodleMoose) [inline test/spec/services/logout.spec.js]: Here

File: test/spec/services/logout.spec.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    cookieValue = 'FACEBOOK';
+    logoutService.logoutFormProviders({ data: { url: 'redirect url' } });
+    expect($window.location.href).toEqual('redirect url');
+  });
+
+  it('should logout from instagram', function() {
```
Comment (TheNoodleMoose) [inline test/spec/services/logout.spec.js]: Here

Comment (TheNoodleMoose) [inline test/spec/services/logout.spec.js]: Here

File: test/spec/services/logout.spec.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    cookieValue = 'INSTAGRAM';
+    logoutService.logoutFormProviders();
+    expect($sce.getTrustedHtml($rootScope.logoutElement)).toEqual(
+      '<iframe class="logout-element" src="https://instagram.com/accounts/logout/" width="0" height="0" ' +
+        "onload=\"document.querySelector('.logout-element').parentNode.removeChild(document.querySelector('.logout-element'));\"/>",
+    );
+  });
+
+  it('should logout from relay', function() {
```
Comment (TheNoodleMoose) [inline test/spec/services/logout.spec.js]: Here.

File: app/scripts/services/logout.js [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+        case 'RELAY':
+          // if relay, then then redirect to the Relay logout URL
+          var serviceUrl = $location.absUrl().replace('logout', '');
```
Comment (TheNoodleMoose) [inline app/scripts/services/logout.js]: could probably be a const instead of var.

File: app/scripts/services/logout.js (hunk 1 above)
Comment (katefree) [reply app/scripts/services/logout.js]: done

File: test/spec/services/logout.spec.js (hunk 2 above)
Comment (katefree) [reply test/spec/services/logout.spec.js]: fixed

File: test/spec/services/logout.spec.js (hunk 3 above)
Comment (katefree) [reply test/spec/services/logout.spec.js]: fixed

File: test/spec/services/logout.spec.js (hunk 4 above)
Comment (katefree) [reply test/spec/services/logout.spec.js]: fixed

File: test/spec/services/logout.spec.js (hunk 5 above)
Comment (katefree) [reply test/spec/services/logout.spec.js]: fixed

File: test/spec/services/logout.spec.js (hunk 6 above)
Comment (katefree) [reply test/spec/services/logout.spec.js]: fixed

File: test/spec/services/logout.spec.js (hunk 7 above)
Comment (katefree) [reply test/spec/services/logout.spec.js]: fixed

File: app/scripts/services/logout.js (hunk 9 above)
Comment (katefree) [reply app/scripts/services/logout.js]: fixed

File: test/spec/services/logout.spec.js (hunk 8 above)
Comment (katefree) [reply test/spec/services/logout.spec.js]: fixed

**Comment by hlbraddock** [comment]
@TheNoodleMoose review requested.

**Pull request merged**